metrics-api/src/main/java/software/amazon/swage/metrics/NullContext.java

```diff
@@ -42,6 +43,13 @@ public void record(Metric label, Number value, Unit unit, Instant time) {
     public void count(Metric label, long delta) {
     }
 
+    @Override
+    public MetricContext newChildContext(TypedMap attributes) {
+        ImmutableTypedMap.Builder childAttributes = ImmutableTypedMap.Builder.from(this.attributes);
+        attributes.stream().forEach(entry -> childAttributes.add(entry.getKey(), entry.getValue()));
```

Comment:
I prefer using a parent pointer here for a few reasons.

I think Metrics Contexts are hierarchical, as such I think the structure should reflect that.
Copying all these values into a new map at this point is slow for the running application.
There's potential for naming collisions, potentially stoping valuable context information from the parent.
Copying all the values defines how contexts are aggregated here, not in the recorder where that aggregation belongs.

Reply:
On the other hand, if children are intended to be completely independent, you could end up with extra context objects hanging around just because childs still refer to them... Think the whole parent/child relationship needs to be codified better before choosing one strategy over the other.